### PR TITLE
Implement `sign_and_broadcast_tx`

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -81,3 +90,98 @@ fn test_drain_empty_wallet() {
...
+
+        let wallet = Wallet::new(Config {
+            electrum_url: "localhost:50000".to_string(),
+            wallet_db_path: ".bdk-database-drain-funds".to_string(),
+            network: Network::Regtest,
+            watch_descriptor: REGTEST_WATCH_DESCRIPTOR.to_string(),
+        })
+        .unwrap();
+
+        try_cmd_repeatedly!(
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply tests/integration_tests.rs]: Cool! Thank you. Applied in [c8bb71c](https://github.com/getlipa/lipa-business-lib/pull/23/commits/c8bb71c59cc9b0e6f29a368edcd4c88f79e6bacd)

File: src/wallet.rs [hunk 2]
```diff
@@ -139,35 +142,45 @@ impl Wallet {
...
+        let is_finalized = wallet
+            .sign(&mut psbt, SignOptions::default())
+            .map_to_permanent_failure("Failed to sign PSBT")?;
+        if !is_finalized {
+            return Err(permanent_failure("Wallet didn't sign all inputs"));
+        }
+
+        self.blockchain
+            .broadcast(&psbt.extract_tx())
+            .map_to_runtime_error(ElectrumServiceUnavailable, "Failed to broadcast tx")
```
Comment (gcomte) [inline src/wallet.rs:167]: If it provides any value, you could return the transaction ID here (see code equivalent in GitLab project)

Comment (danielgranhao) [reply src/wallet.rs:167]: I think it's redundant  here because the `Tx` struct provided by `prepare_drain_tx()` already provides the ID.

Comment (gcomte) [reply src/wallet.rs:167]: Agreed. Let's resolve this conversation

**Pull request merged**